master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -974,11 +976,14 @@ private[celeborn] class Master(
         .asScala.map { case (worker, slots) => worker.toUniqueId -> slots }.asJava,
       requestSlots.requestId)
 
-    logInfo(s"Offer slots successfully for $numReducers reducers of $shuffleKey" +
-      s" on ${slots.size()} workers.")
-
+    var offerSlotsMsg = s"Successfully offered slots for $numReducers reducers of $shuffleKey" +
+      s" on ${slots.size()} workers"
     val workersNotSelected = availableWorkers.asScala.filter(!slots.containsKey(_))
-    val offerSlotsExtraSize = Math.min(conf.masterSlotAssignExtraSlots, workersNotSelected.size)
+    val offerSlotsExtraSize = Math.min(
+      Math.max(
+        slotsAssignExtraSlots,
+        slots.size() - slotsAssignMinWorkers),
```

Comment:
This line contains a logic error. You should use `slotsAssignMinWorkers - slots.size()` to make sure that a shuffle will involve adequate workers.